### PR TITLE
Support for PHP 8

### LINKED ISSUE #31: Support for PHP 8
### What did you do?

Tried to install on `php:8.0-fpm-alpine`

### What did you expect to happen?

Successful install

### What actually happened?

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - nordsoftware/lumen-newrelic[3.0.0, ..., 3.1.1] require php ^7.2 -> your php version (8.0.1) does not satisfy that requirement.
    - Root composer.json requires nordsoftware/lumen-newrelic ^3.0 -> satisfiable by nordsoftware/lumen-newrelic[3.0.0, 3.1.0, 3.1.1].
```

### What versions of relevant language, framework and this module are you using?

Here's the basic `composer.json` I'm using.

```json
{
    "name": "laravel/lumen",
    "description": "The Laravel Lumen Framework.",
    "keywords": ["framework", "laravel", "lumen"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": "^8.0",
        "laravel/lumen-framework": "^8.0",
        "nordsoftware/lumen-newrelic": "^3.0",
        "zircote/swagger-php": "^3.0"
    },
    "require-dev": {
        "fzaninotto/faker": "^1.9.1",
        "phpunit/phpunit": "^9.3",
        "mockery/mockery": "^1.3.1"
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "classmap": [
            "tests/"
        ]
    },
    "scripts": {
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

PHP 8 doesn't bring anything that would break the current version of the package, the upgrade should be straight forward, something like

```json
"require": {
        "php": "^7.2 | ^8.0",
        "intouch/newrelic": "^1.0",
        "illuminate/support": "^5.5 | ^6.0 | ^7.0 | ^8.0",
        "nordsoftware/lumen-chained-exception-handler": "^2.0"
    },
```

May also require an upgrade of the `nordsoftware/lumen-chained-exception-handler` package.

### PR DESCRIPTION
**Pull request by ls-renaud-tilte**
Fixes #31

Changes proposed in this pull request:

 * Update supported PHP version to include PHP 8
 * **Currently blocked by New Relic team: the NewRelic agent for PHP is only working with PHP up to 7.4**
 * **Also blocked by https://github.com/digiaonline/lumen-chained-exception-handler/pull/11**
 
```
  - Installing intouch/newrelic (1.0.3): Extracting archive
  - Installing nordsoftware/lumen-newrelic (dev-php8 260b10b): Extracting archive
  - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
```

### CODE REVIEW

**Comment by MarkG-IEL** [comment]
Status of PHP 8 Support for tracking purposes: https://github.com/orgs/newrelic/projects/23?card_filter_query=milestone%3A%22php+8+support%22

**Comment by ls-renaud-tilte** [comment]
> Status of PHP 8 Support for tracking purposes: https://github.com/orgs/newrelic/projects/23?card_filter_query=milestone%3A%22php+8+support%22

Thanks buddy, the links are added into the issue description
https://github.com/digiaonline/lumen-newrelic/issues/31

**Comment by hugovk** [comment]
> * **Also blocked by [Update composer.json to support PHP8 lumen-chained-exception-handler#11](https://github.com/digiaonline/lumen-chained-exception-handler/pull/11)**

https://github.com/digiaonline/lumen-chained-exception-handler/pull/11 has now been merged and released as 3.0.0.

Please can you update this PR?

**Comment by neclimdul** [comment]
The chained repo merge probably probably didn't affect this but #30 probably did. should be an easy fix though

File: .phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...{"Nord\\Lumen\\NewRelic\\Tests\\NewRelicExceptionHandlerTest::testReportException":4,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicExceptionHandlerTest::testIgnoreNothing":4,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicServiceProviderTest::testRegisterExceptionHandler":3},"times":{"Nord\\Lumen\\NewRelic\\Tests\\NewRelicExceptionHandlerTest::testReportException":0.004,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicExceptionHandlerTest::testReportDefaultIgnoredException":0.001,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicExceptionHandlerTest::testReportIgnoredException":0,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicExceptionHandlerTest::testIgnoreNothing":0,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicMiddlewareTest::testClosureBasedTransactionNames":0.029,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicMiddlewareTest::testNamedRouteTransactionNames":0.001,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicMiddlewareTest::testControllerBasedTransactionNames":0.003,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicMiddlewareTest::testBackgroundJobMiddleware":0.001,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicServiceProviderTest::testRegisterServiceProvider":0.001,"Nord\\Lumen\\NewRelic\\Tests\\NewRelicServiceProviderTest::testRegisterExceptionHandler":0.014}}
```
Comment (neclimdul) [inline .phpunit.result.cache]: Woops?

Comment (ls-renaud-tilte) [reply .phpunit.result.cache]: yes, woops indeed

**Comment by hugovk** [comment]
Thanks, released in 4.0.0! 🚀 

https://packagist.org/packages/nordsoftware/lumen-newrelic#4.0.0

**Pull request merged**